src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/FrameConnection.cs

```diff
@@ -242,10 +267,40 @@ public void Tick(DateTimeOffset now)
 
                 if (_timeoutAction == TimeoutAction.SendTimeoutResponse)
                 {
+                    SetTimeoutResponse();
+                }
+
+                Timeout();
+            }
+            else if (_timingReads)
+            {
+                _readTimingElapsed += timestamp - _lastTimestamp;
+
+                if (_readTimingElapsed > _frame.RequestBodyTimeout.Ticks)
```

Comment:
Ah, those docs are just talking about the rate.

I don't see how an absolute timeout is workable for variable payload sizes. You'd always have to change the size limit and the time limit in tandem, and the time limit is only a multiple of the data rate and the size limit. What value does RequestBodyTimeout add beyond the size and rate limits?